src/firmware/guest/mod.rs

```diff
@@ -37,13 +37,13 @@ impl Firmware {
     pub fn snp_get_report(
         &mut self,
         message_version: Option<u8>,
-        report_request: &mut SnpReportReq,
+        mut report_request: SnpReportReq,
     ) -> Result<AttestationReport, Indeterminate<Error>> {
         let mut report_response: SnpReportRsp = SnpReportRsp::default();
 
         SNP_GET_REPORT.ioctl(
             &mut self.0,
-            &mut SnpGuestRequest::new(message_version, report_request, &mut report_response),
+            &mut SnpGuestRequest::new(message_version, &mut report_request, &mut report_response),
```

Comment:
"_- snp_get_report: the kernel does not write into the request object (and there are no fields for which it would make sense)_"

You still give a `mut` reference here, however. Am I missing something here?

Reply:
No you're note missing anything. The ioctl still requires something potentially mutable to be passed in, but the kernel does not write into the request object for `SNP_GET_REPORT` (the `SnpReportReq` object does not have any output fields). I'm suggesting that we don't expose this internal property to the user, so that they do not need to create a mut variable themselves.